package/MDAnalysis/analysis/base.py

```diff
@@ -124,13 +131,20 @@ def _setup_frames(self, trajectory, start=None, stop=None, step=None):
             stop frame of analysis
         step : int, optional
             number of frames to skip between each analysed frame
+
+
+        .. versionchanged:: 1.0.0
+            Added .frames and .times arrays as attributes
+
         """
         self._trajectory = trajectory
         start, stop, step = trajectory.check_slice_indices(start, stop, step)
         self.start = start
         self.stop = stop
         self.step = step
         self.n_frames = len(range(start, stop, step))
+        self.frames = np.zeros(self.n_frames, dtype=int)
```

Comment:
Would there be any drawback to instead doing:
```
self.frames = np.arange(start, stop, step, dtype=int)
self.times = np.arange(start, stop, step) * trajectory.dt
```
which would remove the need to add these values in `run()`, and also mean that these attributes exist before `run()` is called?

Reply:
I didn't do that for `.times` because I wasn't sure how the `times` were obtained, e.g. I think xtc files must encode them somewhere as actual times instead of time steps because `gmx check` can tell you when you don't have regular time steps. `.frames` is a good idea, but I might see if #2664 is welcome first because I [already stuck it in there as `.frame_indices`. ](https://github.com/MDAnalysis/mdanalysis/pull/2664/files#diff-6d44da489d8c88a50d896dd8df8fd393R133)

Reply:
In general you do not want to compute times yourself. Believe what the reader gives you.

Reply:
I am not in favor because the Reader is responsible for interpreting trajectory data, not the upstream code. Believe what the reader gives you.